ckks/bootstrapping/bootstrap_test.go

```diff
@@ -0,0 +1,142 @@
+package bootstrapping
+
+import (
+	"flag"
+	"fmt"
+	"github.com/ldsec/lattigo/v2/ckks"
+	"github.com/ldsec/lattigo/v2/utils"
+	"github.com/stretchr/testify/assert"
+	"runtime"
+	"testing"
+)
+
+var flagLongTest = flag.Bool("long", false, "run the long test suite (all parameters + secure bootstrapping). Overrides -short and requires -timeout=0.")
+var testBootstrapping = flag.Bool("test-bootstrapping", false, "run the bootstrapping tests (memory intensive)")
+
+func ParamsToString(params ckks.Parameters, opname string) string {
+	return fmt.Sprintf("%slogN=%d/LogSlots=%d/logQP=%d/levels=%d/a=%d/b=%d",
+		opname,
+		params.LogN(),
+		params.LogSlots(),
+		params.LogQP(),
+		params.MaxLevel()+1,
+		params.PCount(),
+		params.Beta())
+}
+
+func TestBootstrapParametersMarshalling(t *testing.T) {
+	bootstrapParams := DefaultParameters[0]
+	data, err := bootstrapParams.MarshalBinary()
+	assert.Nil(t, err)
+
+	bootstrapParamsNew := new(Parameters)
+	if err := bootstrapParamsNew.UnmarshalBinary(data); err != nil {
+		assert.Nil(t, err)
+	}
+	assert.Equal(t, bootstrapParams, *bootstrapParamsNew)
+}
+
+func TestBootstrap(t *testing.T) {
+
+	if runtime.GOARCH == "wasm" {
+		t.Skip("skipping bootstrapping tests for GOARCH=wasm")
+	}
+
+	if !*testBootstrapping {
+		t.Skip("skipping bootstrapping tests (add -test-bootstrapping to run the bootstrapping tests)")
+	}
+
+	paramSet := 0
+
+	ckksParams := DefaultCKKSParameters[paramSet]
+	bootstrapParams := DefaultParameters[paramSet]
+
+	// Insecure params for fast testing only
+	if !*flagLongTest {
+		ckksParams.LogN = 14
+		ckksParams.LogSlots = 13
+	}
+
+	// Tests homomorphic modular reduction encoding and bootstrapping on sparse slots
+	params, err := ckks.NewParametersFromLiteral(ckksParams)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, testSet := range []func(params ckks.Parameters, btpParams Parameters, t *testing.T){
```

Comment:
Should simplify the main loop now that other functionalities are tested in advanced package.

Reply:
update comments